protocol/v1/status_message.go

```diff
@@ -147,26 +163,17 @@ func (m *StatusMessage) HandleApplicationMetadata() error {
 }
 
 func (m *StatusMessage) HandleApplication() error {
-	value, err := decodeTransitMessage(m.DecryptedPayload)
+	// Try protobuf first
+	message := &protobuf.ChatMessage{}
+
+	err := proto.Unmarshal(m.DecryptedPayload, message)
 	if err != nil {
-		log.Printf("[message::DecodeMessage] could not decode message: %#x, err: %v", m.Hash, err.Error())
-		return err
-	}
-	m.ParsedMessage = value
-	switch m.ParsedMessage.(type) {
-	case Message:
+		log.Printf("[message::DecodeMessage] could not decode protobuf message: %#x, err: %v", m.Hash, err.Error())
+	} else {
 		m.MessageType = MessageT
-	case MembershipUpdateMessage:
-		m.MessageType = MembershipUpdateMessageT
-	case PairMessage:
-		m.MessageType = PairMessageT
-		// By default we null the parsed message field, as
-		// otherwise is populated with the raw transit and we are
-		// unable to marshal in case it contains maps
-		// as they have type map[interface{}]interface{}
-	default:
-		m.ParsedMessage = nil
+		m.ParsedMessage = message
```

Comment:
This way here we will have `nil` instead of a pointer to an empty struct. Not sure if this changes much but later it can be checked with `m.ParsedMessage != nil` which might be useful.